runtime/src/main/java/org/corfudb/runtime/BootstrapUtil.java

```diff
@@ -35,12 +37,51 @@
      * @param retries      Number of retries to bootstrap each node before giving up.
      * @param retryTimeout Duration between retries.
      */
-    public static void bootstrap(@Nonnull Layout layout,
+    public static void bootstrap(@NonNull Layout layout,
                                  int retries,
-                                 @Nonnull Duration retryTimeout) {
+                                 @NonNull Duration retryTimeout) {
         bootstrap(layout, CorfuRuntimeParameters.builder().build(), retries, retryTimeout);
     }
 
+    /**
+     * Bootstraps a layout server connected to the specified router.
+     *
+     * @param router Router connected to the node.
+     * @param layout Layout to bootstrap with
+     */
+    private static void bootstrapLayoutServer(IClientRouter router, Layout layout)
+            throws ExecutionException, InterruptedException, AlreadyBootstrappedException {
+        LayoutClient layoutClient = new LayoutClient(router, layout.getEpoch());
+
+        try {
+            CFUtils.getUninterruptibly(layoutClient.bootstrapLayout(layout),
+                    AlreadyBootstrappedException.class);
+        } catch (AlreadyBootstrappedException abe) {
+            if (!layoutClient.getLayout().get().equals(layout)) {
+                log.error("Layout Server already bootstrapped with different layout.");
+                throw abe;
+            }
+        }
+    }
+
+    /**
+     * Bootstraps a management server connected to the specified router.
+     *
+     * @param router Router connected to the node.
+     * @param layout Layout to bootstrap with
+     */
+    private static void bootstrapManagementServer(IClientRouter router, Layout layout) {
+        ManagementClient managementClient
+                = new ManagementClient(router, layout.getEpoch());
+
+        try {
+            CFUtils.getUninterruptibly(managementClient.bootstrapManagement(layout),
+                    AlreadyBootstrappedException.class);
+        } catch (AlreadyBootstrappedException abe) {
+            // Ignore Exception if already bootstrapped.
```

Comment:
Add some log info here?

Reply:
Done.

Reply:
Hmm, stale code? This still doesn't have a log.